src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs

```diff
@@ -239,7 +266,7 @@ public virtual TResult VisitGroupExpressionNode(GroupExpressionNode node)
 
```

Comment:
Is this expected in `class AssociativeAstVisitor<TResult>`:

```csharp
public virtual TResult Visit(Node node)
{
    var assocNode = node as AssociativeNode;
    if (assocNode != null)
        return assocNode.Accept(this);

    return default(TResult);
}
```

Should the `AssociativeNode` be replaced by `TResult` here since it's templated to fit `TResult`?